### PR TITLE
prow-job-dispatcher: Add flag -prometheus-days-before

### PR DESCRIPTION
**Pull request by hongkailiu**
/cc @alvaroaleman @stevekuznetsov

### CODE REVIEW

File: pkg/dispatcher/prometheus.go [hunk 1]
```diff
@@ -30,8 +30,9 @@ type PrometheusAPI interface {
 }
 
 // GetJobVolumesFromPrometheus gets job volumes from a Prometheus server
-func GetJobVolumesFromPrometheus(ctx context.Context, prometheusAPI PrometheusAPI) (map[string]float64, error) {
-	result, warnings, err := prometheusAPI.Query(ctx, `sum(increase(prowjob_state_transitions{state="pending"}[7d])) by (job_name)`, time.Now())
+func GetJobVolumesFromPrometheus(ctx context.Context, prometheusAPI PrometheusAPI, daysBefore int) (map[string]float64, error) {
```
Comment (petr-muller) [inline pkg/dispatcher/prometheus.go:33]: `daysBefore` is not used?

File: pkg/dispatcher/prometheus.go [hunk 2]
```diff
@@ -30,8 +30,9 @@ type PrometheusAPI interface {
...
+	y, m, d := time.Now().Add(-24 * time.Hour).Date()
```
Comment (petr-muller) [inline pkg/dispatcher/prometheus.go:34]: `Sub(24)` instead of `Add(-24)`

File: pkg/dispatcher/prometheus.go (hunk 1 above)
Comment (hongkailiu) [reply pkg/dispatcher/prometheus.go:33]: Updated.

File: pkg/dispatcher/prometheus.go [hunk 3]
```diff
@@ -30,8 +30,9 @@ type PrometheusAPI interface {
 }
 
 // GetJobVolumesFromPrometheus gets job volumes from a Prometheus server
-func GetJobVolumesFromPrometheus(ctx context.Context, prometheusAPI PrometheusAPI) (map[string]float64, error) {
-	result, warnings, err := prometheusAPI.Query(ctx, `sum(increase(prowjob_state_transitions{state="pending"}[7d])) by (job_name)`, time.Now())
+func GetJobVolumesFromPrometheus(ctx context.Context, prometheusAPI PrometheusAPI, daysBefore int) (map[string]float64, error) {
+	y, m, d := time.Now().Add(-time.Duration(24*(daysBefore+1)) * time.Hour).Date()
```
Comment (petr-muller) [inline pkg/dispatcher/prometheus.go:34]: The `+1` makes the interface of this function confusing, and thinking about it, "days before yesterday" is confusing too. How about make this a 1..15 interval (disallow 0), with `1` being "last day" and `15` "last 15 days", then both the semantics and code can be simpler.

File: cmd/prow-job-dispatcher/main.go [hunk 4]
```diff
@@ -48,7 +48,7 @@ func gatherOptions() options {
 	fs.StringVar(&o.prometheusURL, "prometheus-url", "https://prometheus-prow-monitoring.apps.ci.l2s4.p1.openshiftapps.com", "The prometheus URL")
 	fs.StringVar(&o.prometheusUsername, "prometheus-username", "", "The Prometheus username.")
 	fs.StringVar(&o.prometheusPasswordPath, "prometheus-password-path", "", "The path to a file containing the Prometheus password")
-	fs.StringVar(&o.jobVolumesPath, "job-volumes-path", filepath.Join(os.TempDir(), "job-volumes.yaml"), "The path to a file containing the job volumes")
+	fs.IntVar(&o.prometheusDaysBefore, "prometheus-days-before", 0, "Number [0,15] of days before yesterday. 00-00-00 of that day will be used as time to query Prometheus. E.g., 0 means 00-00-00 yesterday.")
```
Comment (petr-muller) [inline cmd/prow-job-dispatcher/main.go:51]: days of *what* before yesterday? "Use last X days of metrics from Prometheus".

File: pkg/dispatcher/prometheus.go (hunk 3 above)
Comment (hongkailiu) [reply pkg/dispatcher/prometheus.go]: Move the calculation of time out of the function.
It is always the last 7 days (part of the query).
The time cal is for given a time as now for the last 7 days.

File: cmd/prow-job-dispatcher/main.go (hunk 4 above)
Comment (hongkailiu) [reply cmd/prow-job-dispatcher/main.go]: See if it is better now.

**Review by petr-muller — APPROVED** [review]

**Pull request merged**